### ISSUE TITLE
examples are broken due to Yahoo

### ISSUE THREAD

**Issue by xiaoronglv** [ruby-concurrency/concurrent-ruby#670]
Lots of examples are requesting yahoo finance, but this api is discontinued by Yahoo.

so the following examples doesn't work for newbies. Could you please replace yahoo?

Many thanks.
 
```
require 'concurrent'
require 'thread'   # for Queue
require 'open-uri' # for open(uri)

class Ticker
  def get_year_end_closing(symbol, year)

    uri = "http://ichart.finance.yahoo.com/table.csv?s=#{symbol}&a=11&b=01&c=#{year}&d=11&e=31&f=#{year}&g=m"
    puts uri
    data = open(uri) {|f| f.collect{|line| line.strip } }
    data[1].split(',')[4].to_f
  end
end
```

## Reference

https://forums.yahoo.net/t5/Yahoo-Finance-help/Is-Yahoo-Finance-API-broken/td-p/250503

## URI

http://ichart.finance.yahoo.com/table.csv?s=TWTR&a=11&b=01&c=2013&d=11&e=31&f=2013&g=m

**Comment by pitr-ch** [comment]
Thanks, we'll fix it.

**Comment by GustavoCaso** [comment]
I'm going to have a look at this one 😄

**Comment by pitr-ch** [comment]
Thanks, @GustavoCaso!

**Comment by pitr-ch** [comment]
@GustavoCaso are you still looking into this? If not please let me know.

**Comment by GustavoCaso** [comment]
Hey @pitr-ch, first of all, thank you for the invite. Sorry for the huge delay swamp with work and other contributions. But count with me for this one

**Comment by pitr-ch** [comment]
Ok great, I've assigned it to you.

**Comment by GustavoCaso** [comment]
Thanks

**Comment by GustavoCaso** [comment]
@pitr-ch I started working on fixing the examples but got stack when the example works only if I `puts` the row.

Here is a gist with the example and a comment, any feedback is very welcome https://gist.github.com/GustavoCaso/862139e82718341db4cfb00ea8720f2e

**Comment by pitr-ch** [comment]
Thanks for looking at it, sorry I missed your comment. The gist example works for me without `puts` added, how does it fail for you?

**Comment by GustavoCaso** [comment]
@pitr-ch If I run it multiple times there is no output, maybe the first time it is, but later there is no output

**Comment by pitr-ch** [comment]
@GustavoCaso try to call `#value!` instead of `value` it will give you an error happening in `get_year_end_closing`. The error is there even with the puts added. Btw the example does not need to use the same API you can do something with GitHub API or anything else if you like.

**Comment by pitr-ch** [comment]
Just got to looking at your PR :)

**Comment by chrisseaton** [comment]
Fixed in #937.
